### PR TITLE
progress #29 documentation for Tango on Docker

### PR DESCRIPTION
**Pull request by Ingvord**
New PR for issue #29

### CODE REVIEW

File: source/advanced/testing-tango-using-docker.rst [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+https://github.com/tango-controls/tango-cs
+
+[inject Dockerfile]
+
+[header] Tango docker stack
+
+Typical tango stack looks like this:
+
+|image/tango-stack.png|
```
Comment (piogor) [inline source/advanced/testing-tango-using-docker.rst]: .. image:: testing-using-tango-docker/tango-stack.png

File: source/advanced/testing-tango-using-docker.rst [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+These containers can be found in the corresponding repositories:
+https://github.com/tango-controls/docker-mysql
+
+[inject Dockerfile]
+
+https://github.com/tango-controls/tango-cs
+
+[inject Dockerfile]
+
+[header] Tango docker stack
```
Comment (piogor) [inline source/advanced/testing-tango-using-docker.rst]:
```
Tabgo dockec
~~~~~~~~~
```

or:

```
.. rubric::
 
   Tango docker stach
```

File: source/advanced/testing-tango-using-docker.rst [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+Once docker containers are up and running one can access mysql on localhost:9999 and the tango host on localhost:10000
+
+For instance, one can start jive (assuming it is installed on the system) [link to jive] :command:`TANGO_HOST=localhost:10000 jive`
+
+[img?]
+
+[header] Tango docker stack for Tango REST API
+
+One can setup Tango docker stack for Tango REST API [reference] as well. Currently only mtangorest.server [link] provides docker container that can be used together with Tango docker containers:
```
Comment (piogor) [inline source/advanced/testing-tango-using-docker.rst]: Put in the Tango REST API document an anchor:

.. _tango_rest_api:

Your text..

Then use :ref:`tango_rest_api` to provide a link

File: source/advanced/testing-tango-using-docker.rst [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+Applying additional SQL script to tangocs/mysql.
+
+Since every docker image can be used as a base for another docker image one can create his own image. In this new image new SQL scripts can be applied to extend the exiting scheme, for instance to add new devices or adjust configuration.
+
+An example of such extension could look like this:
+
+[TODO script]
+
+
+
```
Comment (piogor) [inline source/advanced/testing-tango-using-docker.rst]: For including exrernal file you may add a download command in conf.py then use standard

.. include:: path/to/file

**Comment by Ingvord** [comment]
I will add the {TODO acript} later on after testing everything

**Comment by Ingvord** [comment]
@piogor could you please merge this PR?

**Comment by piogor** [comment]
I will do it 😊.

**Comment by piogor** [comment]
Hi, in addition to moving to a new location I have cropped the tango-stack.png image, since it was making a distance in text

**Comment by piogor** [comment]
Hi @Ingvord, the Travis is complaining about a link to tango-cs container:
```
writing output... [ 20%] development/debugging-and-testing/testing-tango-using-docker

(line  109) broken    http://localhost:10001/tango/rest - HTTPConnectionPool(host='localhost', port=10001): Max retries exceeded with url: /tango/rest (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f6dd7f4a450>: Failed to establish a new connection: [Errno 111] Connection refused',))

(line   24) broken    https://github.com/tango-controls/tango-cs - 404 Client Error: Not Found for url: https://github.com/tango-controls/tango-cs
```
Shell I merge it anyway?

**Comment by Ingvord** [comment]
@piogor should fixed in 249371e
@chedburgh thanks Stuart!

**Comment by piogor** [comment]
Thanks @Ingvord. I will aslo add `linkcheck_ignore` to conf.py to ignore `http://localhost*` patterns to prevent Travis from misleading and then do merge.

**Comment by piogor** [comment]
@Ingvord, if you do not need the branch for future you may delete it.

**Pull request merged**